.NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs

```diff
@@ -311,26 +311,26 @@ private static DateTimeResolutionResult ParseYearToYear(string text, DateObject
                     }
                 }
 
-                if (beginYear < 100 && beginYear >= 90)
+                if (beginYear < 100 && beginYear >= 50)
                 {
                     beginYear += 1900;
                 }
-                else if (beginYear < 100 && beginYear < 20)
+                else if (beginYear < 100 && beginYear < 50)
                 {
                     beginYear += 2000;
                 }
 
-                if (endYear < 100 && endYear >= 90)
+                if (endYear < 100 && endYear >= 50)
                 {
                     endYear += 1900;
                 }
-                else if (endYear < 100 && endYear < 20)
+                else if (endYear < 100 && endYear < 50)
                 {
                     endYear += 2000;
                 }
 
                 var beginDay = DateObject.MinValue.SafeCreateFromValue(beginYear, 1, 1);
-                var endDay = DateObject.MinValue.SafeCreateFromValue(endYear, 12, 31);
+                var endDay = DateObject.MinValue.SafeCreateFromValue(endYear, 1, 1);

```

Comment:
Why this change? The end day of a typical year range is the last day of the last year.

Reply:
See comment below. "this year" should correctly be
start: 2018-01-01
end: 2019-01-01